buildpack/telemetry/appdynamics_telegraf_output.py

```diff
@@ -0,0 +1,132 @@
+"""The module is used in runtime to convert metrics from telegraf for appdynamics Machine Agent"""
+import json
+import logging
+import requests
+from requests.exceptions import ConnectionError
+
+
+METRIC_TREE_BASE_NODE = "Custom Metrics|Mx Runtime Statistics"
+AGGR_TYPE = "OBSERVATION"
+MACHINE_AGENT_URL = "http://127.0.0.1:8293/api/v1/metrics"
+
+
+def _filter_last(payload):
+    """
+    The function retrieves from the payload the latest version of
+    each metric. It is required since the payload probably can
+    contain buffered metrics. But Machine Agent HTTP listener payload
+    doesn't support any timestamps.
+
+    """
+
+    last_metrics = {}
+
+    for metric in payload:
+
+        metric_name = metric["metricName"]
+        last_metric = last_metrics.get(metric_name)
+
+        if last_metric:
+            if metric["timestamp"] > last_metric["timestamp"]:
+                last_metrics[metric_name] = metric
+        else:
+            last_metrics[metric_name] = metric
+
+    filtered_metrics = []
+
+    for metric in last_metrics.values():
+        del metric["timestamp"]
+        filtered_metrics.append(metric)
+
+    return filtered_metrics
+
+
+def _convert_metric(metric):
+
+    fields = metric.get("fields")
+
+    converted = []
+
+    if fields is None:
+        logging.error("Invalid format of specific metric (telegraf)")
+        return
+
+    for value_name in fields.keys():
+        value = fields[value_name]
+        if value_name == "value":
+            metric_name = metric["name"]
+        else:
+            metric_name = "_".join((metric["name"], value_name))
+
+        metric_path_list = [METRIC_TREE_BASE_NODE, metric_name]
+
+        tags = metric.get("tags")
+        if tags:
+            area = tags.get("area")
+            id = tags.get("id")
+            db = tags.get("db")
+
+            if area:
+                metric_path_list.append(area)
+            if id:
+                metric_path_list.append(id)
+            if db:
+                metric_path_list.append(db)
```

Comment:
This would show up as some unique id. Would that be apparent enough ?
How about we prefix/suffix the id with the term `database` ?

Reply:
Made this: Database '<db_id>'